tests/integration/compact_test.py

```diff
@@ -0,0 +1,141 @@
+import typing
+
+from mock.mock import patch
+
+import hazelcast
+from hazelcast.compact import CompactSchemaService
+from hazelcast.errors import IllegalStateError
+from hazelcast.future import ImmediateFuture, ImmediateExceptionFuture
+from hazelcast.serialization.api import CompactSerializer, CompactWriter, CompactReader
+from tests.base import HazelcastTestCase
+from tests.util import random_string
+
+
+# The tests under this file are meant to be not
+# run with the backward compatibility tests, as
+# they test implementation details through private
+# APIs. If you are going to add a test to this file,
+# make sure that it fits the description above. If not,
+# add it to tests under the backward_compatible folder.
+
+
+class CompactSchemaReplicationRetryTest(HazelcastTestCase):
+    rc = None
+    cluster = None
+    retry_count = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.rc = cls.create_rc()
+        cls.cluster = cls.create_cluster(cls.rc, None)
+        cls.cluster.start_member()
+        cls.cluster.start_member()
+        cls.retry_count = CompactSchemaService._SEND_SCHEMA_RETRY_COUNT
+        CompactSchemaService._SEND_SCHEMA_RETRY_COUNT = 10
+
+    def setUp(self) -> None:
+        self.client = hazelcast.HazelcastClient(
+            compact_serializers=[FooSerializer()],
+            cluster_name=self.cluster.id,
+            invocation_retry_pause=0.1,
+        )
+        self.map = self.client.get_map(random_string()).blocking()
+
+    def tearDown(self) -> None:
+        self.client.shutdown()
+
+    @classmethod
+    def tearDownClass(cls):
+        CompactSchemaService._SEND_SCHEMA_RETRY_COUNT = cls.retry_count
+        cls.rc.terminateCluster(cls.cluster.id)
+        cls.rc.exit()
+
+    def test_compact_schemas_replicated(self):
```

Comment:
Could you add a reference to where the test is coming from in case this is a port of a Java test?